tools/adaparse.cpp

```diff
@@ -38,33 +177,68 @@
  *      | `------------------------------ username_end 7
  *      `-------------------------------- protocol_end 5
  **/
+
 int main(int argc, char** argv) {
+  std::ios::sync_with_stdio(false);
+
   cxxopts::Options options("adaparse",
                            "Command-line version of the Ada URL parser");
 
   options.add_options()("d,diagram", "Print a diagram of the result",
                         cxxopts::value<bool>()->default_value("false"))(
       "u,url", "URL Parameter (required)", cxxopts::value<std::string>())(
-      "h,help", "Print usage");
+      "h,help", "Print usage")(
+      "g,get", "Get a specific part of the URL (e.g., 'origin', 'host', etc.)",cxxopts::value<std::string>())(
+      "b,benchmark", "Run benchmark for piped file functions",cxxopts::value<bool>()->default_value("false") )
+      //"p,path", "Takes in a path and process all the files within", cxxopts::value<std::string>())(
+      
+      ;
   options.parse_positional({"url"});
 
   auto result = options.parse(argc, argv);
+#ifdef _MSC_VER
+  if (!_isatty(_fileno(stdin))) {
+#else
+  if (!isatty(fileno(stdin))) {
+#endif
+  return piped_file(result,std::cin) ? EXIT_SUCCESS : EXIT_FAILURE;
+
+  }
+  
   // the first argument without an option name will be parsed into file
   if (result.count("help") || !result.count("url")) {
-    std::cout << options.help() << std::endl;
+    fmt::print(stderr, "{}\n", options.help());
     return EXIT_SUCCESS;
   }
+
+
+/* To be completed later? 
```

Comment:
I would drop this for now... consider deleting this comment.

Reply:
Or maybe keep it, but using FILE* descriptors that default on std::in might be great.